modules/custom/wim_cvdr/wim_cvdr.module

```diff
@@ -0,0 +1,235 @@
+  $blocks = array();
+  $subjects = _wim_cvdr_retrieve_subjects();
+
+  // Create a seperate block for each subject.
+  foreach ($subjects as $subject) {
+    // Create nice machine name based on subject name.
+    $machine_name = str_replace('-', '_', drupal_html_class($subject));
+    $blocks[md5('wim_cvdr_' . $machine_name)] = array(
+      'info' => t('CVDR: !subject', array('!subject' => $subject)),
+      'cache' => DRUPAL_CACHE_GLOBAL,
+    );
+  }
+
+  return $blocks;
+}
+
+/**
+ * Implements hook_block_view().
+ */
+function wim_cvdr_block_view($delta = '') {
+  $block = array();
+
+  $subjects = _wim_cvdr_retrieve_subjects();
+  foreach ($subjects as $subject) {
+    // Create nice machine name based on subject name.
+    $machine_name = str_replace('-', '_', drupal_html_class($subject));
+
+    // Check if block is current subject.
+    if ($delta === md5('wim_cvdr_' . $machine_name)) {
+      // Render subject block.
+      $block['subject'] = $subject;
+
+      // Build the links to the CVDR items.
+      $items = _wim_cvdr_retrieve_items($subject);
+
+      // Sort the items alphabetically.
+      sort($items);
+
+      foreach ($items as $item) {
+        $list_items[] = l($item['title'], $item['url'],
+          array('attributes' => array('target' => '_blank')));
+      }
+
+      $block['content'] = theme('item_list', array(
+        'items' => $list_items,
+        'title' => '',
+        'type' => 'ul',
+        'attributes' => array(
+          'class' => array('cvdr-wim', drupal_html_class($subject)),
+        ),
+      ));
+    }
+  }
+
+  // Add JS.
+  drupal_add_js(array(
+    'wim_cvdr' => array(
+      'maxItems' => variable_get('wim_cvdr_max_visible', 5) - 1,
+    ),
+  ), 'setting');
+  drupal_add_js(
+    drupal_get_path('module', 'wim_cvdr') . '/wim_cvdr.js', array(
+      'type' => 'file',
+      'scope' => 'footer',
+    )
+  );
+
+  return $block;
+}
+
+/**
+ * Retrieve the subjects available from the webservice.
+ *
+ * @return array
+ *   containing the subjects.
+ */
+function _wim_cvdr_retrieve_subjects() {
+  $items = array();
+
+  // Use caching to improve performance.
+  if ($cached = cache_get('wim_cvdr_items')) {
+    $items = $cached->data;
+  }
+  else {
+    $xml = _wim_cvdr_read_xml();
+    if ($xml) {
+      $items = _wim_cvdr_process_xml($xml);
+    }
+  }
+
+  return array_keys($items);
+}
+
+/**
+ * Retrieve all items from the webservice.
+ *
+ * @param string $key
+ *   Indicating which block to render.
+ *
+ * @return array
+ *   Containing all items.
+ */
+function _wim_cvdr_retrieve_items($key = '') {
+  $items = array();
+
+  // Use caching to improve performance.
+  if ($cached = cache_get('wim_cvdr_items')) {
+    $items = $cached->data;
+  }
+  else {
+    $xml = _wim_cvdr_read_xml();
+    if ($xml) {
+      $items = _wim_cvdr_process_xml($xml);
+    }
+  }
+
+  // Return specific items if requested.
+  if (!empty($key) && array_key_exists($key, $items)) {
+    return $items[$key];
```

Comment:
This returns a single item instead of an array

Reply:
in this case we have an array in output